proxy/proxy.go

```diff
@@ -0,0 +1,173 @@
+package proxy
+
+import (
+	"fmt"
+	"net/http"
+	"net/http/httputil"
+	"net/url"
+	"regexp"
+	"strings"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+	jwt "github.com/dgrijalva/jwt-go"
+)
+
+type Proxy struct {
+	jwtSecret string
+	apis      []*ApiProxy
+}
+
+type ApiProxy struct {
+	name    string
+	matcher *regexp.Regexp
+	handler http.Handler
+	token   string
+	roles   []string
+}
+
+type JWTClaims struct {
+	ID           string                 `json:"id"`
+	Email        string                 `json:"email"`
+	AppMetaData  map[string]interface{} `json:"app_metadata"`
+	UserMetaData map[string]interface{} `json:"user_metadata"`
+	*jwt.StandardClaims
+}
+
+// Error is an error with a message
+type Error struct {
+	Message string `json:"msg"`
+	Code    int    `json:"code"`
+}
+
+var bearerRegexp = regexp.MustCompile(`^(?i)Bearer (\S+$)`)
+
+func New(secret string, apis []*ApiProxy) *Proxy {
+	return &Proxy{
+		jwtSecret: secret,
+		apis:      apis,
+	}
+}
+
+func NewApiProxy(name, urlString, token string, roles []string) (*ApiProxy, error) {
+	matcher := regexp.MustCompile("^/" + name + "/?")
```

Comment:
instead of using the `MustCompile`, I think it would make sense to use the `Compile` and error check. This way we could surface that an not panic